packages/encryption/src/ec.ts

```diff
@@ -101,6 +109,58 @@ function sharedSecretToKeys(sharedSecret: Buffer): { encryptionKey: Buffer; hmac
   };
 }
 
+/**
+ * @ignore
+ */
+function allHexChars(maybe: string): boolean {
+  return maybe.match(/^[0-9a-f]+$/i) !== null;
+}
+
+/**
+ * @ignore
+ */
+function isValidPublicKey(pub: string): {
+  result: boolean;
+  reason: string | null;
+  reason_data: string | null;
+} {
+  const invalidFormat = {
+    result: false,
+    reason_data: 'Invalid public key format',
+    reason: InvalidPublicKeyReason.InvalidFormat,
+  };
+  const invalidPoint = {
+    result: false,
+    reason_data: 'Public key is not a point',
+    reason: InvalidPublicKeyReason.IsNotPoint,
+  };
+  if (pub.length !== 66 && pub.length !== 130) return invalidFormat;
+
+  const firstByte = pub.slice(0, 2);
+
+  // uncompressed public key
+  if (pub.length === 130 && firstByte !== '04') return invalidFormat;
+
+  // compressed public key
+  if (pub.length === 66 && firstByte !== '02' && firstByte !== '03') return invalidFormat;
```

Comment:
This looks right to me. Could you add test cases for all these branches? Well known valid keys for each version/prefix, and invalid keys for each?

Reply:
@zone117x Test cases added for all branches.